proxy/proxy.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"os"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// ProxyEnvNames are standard environment variables for proxies
+var ProxyEnvNames = []string{"HTTP_PROXY", "HTTPS_PROXY", "NO_PROXY"}
+
+// ReadProxyVarsFromEnv retrieves the standard proxy-related environment
+// variables from the running environment and returns a slice of corev1 EnvVar
+// containing upper and lower case versions of those variables.
+func ReadProxyVarsFromEnv() []corev1.EnvVar {
+	envVars := []corev1.EnvVar{}
+	for _, s := range ProxyEnvNames {
+		value := os.Getenv(s)
```

Comment:
does this mean that its upto the user of this library to verify if the variable is empty or not? In the sense, even if the variables are not set, the value is appended in `[]corev1.EnvVar`. Do we need to add a check to omit the ones which are not set, or was this intentional ?

Reply:
Good catch, I was just setting it to empty string in that case, but I can use `LookupEnv` to only pass through only when the var is set.